tests/unit_tests/scheduler/test_lsf_driver.py

```diff
@@ -267,20 +267,34 @@ async def test_kill(
     bkill_path.write_text(
         f"#!/bin/sh\necho '{bkill_stdout}'\n"
         f"echo '{bkill_stderr}' >&2\n"
-        f"echo $@ > 'bkill_args'\n"
+        f"echo $@ >> 'bkill_args'\n"
         f"exit {bkill_returncode}",
         encoding="utf-8",
     )
     bkill_path.chmod(bkill_path.stat().st_mode | stat.S_IEXEC)
 
     driver = LsfDriver()
     driver._iens2jobid = mocked_iens2jobid
+    driver._sleep_time_between_bkills = 0
+
     await driver.kill(iens_to_kill)
+
+    async def wait_for_sigkill_in_file():
+        while True:
+            bkill_args = (
+                Path("bkill_args").read_text(encoding="utf-8").strip().split("\n")
+            )
+            if f"-s SIGKILL {mocked_iens2jobid[iens_to_kill]}" in bkill_args:
```

Comment:
Not sure if this a good way to do it - it might just never return if the correct jobid is not present. What about just assert / return true/false for the jobid after we got SIGGILL in the file?

Reply:
Yes, I will change it.